Should we ask people to raise an RFC prior to making changes to the platform? (#6185)
guardian/dotcom-rendering: .github/ISSUE_TEMPLATE/request-for-change.md

---
name: DCR Request for change
about: Tell us about a feature that you're working on
title: "[RFC]"
labels: RFC, dotcom-rendering
assignees: ''

---

> The Dotcom team would like to help you with your proposed change. Please provide us with a brief and concise explanation for what you're looking to do we will try to make the process easier with early feedback.

Reviewer: Is it worth indicating the type of change an RFC is necessary for? In principal this seems like a great idea, but it would be good to lay down the scale of change this applies for. 

There are also other teams that are part responsible for some code in dotcom-rendering (thinking commercial / T&C etc.) Would BAU changes to code those teams are concerned with require an RFC, i'd guess not but worth laying it out?

Author: This is a good point. I don't think anyone wants to be asked to jump through hoops when making changes to their 'own' code so we need to be clear about when this might or might now apply.

I think in reality what we will end up on is asking people to use their own best judgement. We're a clever bunch here at the Guardian and, more than that, we're very good at working autonomously and making the right choices and so maybe am optional approach would work. But equally, there are benefits to automation taking over and reducing the amount of thinking engineers need to do. What do you think?

Reviewer: > I think in reality what we will end up on is asking people to use their own best judgement.

This makes sense, but for new folk it's probably worth being clear (perhaps with examples) in the template about when an RFC would be appreciated.

Author: 👍  I've added more information to the description to make it clearer about when this RFC should apply

> **Who is responsible for completing this RFC process?**

Reviewer: Will this change be enforced by repository rules / codeowners configuration, or will we rely on Dotcom team members commenting on PRs and nudging them towards the RFC process?

Author: Initially, we should keep the flow optional and only start to enforce things as we learn more about usage

> Once the RFC is raised, it is the responsibility of the _Dotcom team_ to complete this process

Reviewer 2: ```suggestion
Once the RFC is raised, it is the responsibility of the _Dotcom team_ to provide feedback and help you get your change to the finish line.
```

Maybe? I feel like it currently implies that we're going to be the ones executing the RFC!

---
name: DCR Request for change
about: Tell us about a feature that you're planning to implement in dotcom rendering
title: "[RFC]"
labels: RFC, dotcom-rendering
assignees: ''

---

The Dotcom team would like to help you with your proposed change. Please provide us with a brief and concise explanation for what you're looking to do we will try to make the process easier with early feedback or advice.

**When should this RFC be used?**
This is an optional process. If you own the code you're working on or feel confident about the change you're making then you should simply raise a PR as normal. We want developers and teams to feel empowered to make the choice about when to reach out.

> But if your change affects core platform code or is introducing a new concept or pattern, or if you're just looking for more eyes and input then this process is for you.

Reviewer 3: I love a “but”, but wonder if this reads nicer?

```suggestion
However, if your change affects core platform code or is introducing a new concept or pattern, or if you're just looking for more eyes and input then this process is for you.
```

Author: Yep

**Who is responsible for completing this RFC process?**

> Once the RFC is raised, it is the responsibility of the _Dotcom team_ to provide feedback and help you get your change to the finish line.

Reviewer 3: ```suggestion
Once the RFC is raised, it is the responsibility of the _Dotcom team_ (@guardian/dotcom-platform) to provide feedback and help you get your change to the finish line.
```

Author: Better
